### PR TITLE
added docstrings for teamEventAPI.ts

### PR DESCRIPTION
**Pull request by oscarwang20**
### Summary 

- Added docstrings for the endpoints in `teamEventAPI.ts`

### Notes 

- Some of the specification will need to be updated after https://github.com/cornell-dti/idol/pull/521 gets merged

### CODE REVIEW

**Review by kevinmram — APPROVED** [review]
LGTM! (rerun format/prettier to fix build error?)

File: backend/src/API/teamEventsAPI.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@ import PermissionsManager from '../utils/permissionsManager';
 
 const teamEventAttendanceDao = new TeamEventAttendanceDao();
 
+/**
+ * Gets all team events and their attendees and requests. If the user is not a
+ * lead or admin, then an error is thrown.
+ * @param user
```
Comment (andrew032011) [inline backend/src/API/teamEventsAPI.ts:11]: Description for `user`?

**Review by andrew032011 — CHANGES_REQUESTED** [review]
Can you fix the params to be formatted like this?
```
 * @param teamEventInfo - <description>
```

File: backend/src/API/teamEventsAPI.ts (hunk 1 above)
Comment (oscarwang20) [reply backend/src/API/teamEventsAPI.ts]: Done!

File: backend/src/API/teamEventsAPI.ts [hunk 2]
```diff
@@ -5,6 +5,12 @@ import PermissionsManager from '../utils/permissionsManager';
 
 const teamEventAttendanceDao = new TeamEventAttendanceDao();
 
+/**
+ * Gets all team events and their attendees and requests. If the user is not a
+ * lead or admin, then an error is thrown.
+ * @param user - the user submitting the request -
```
Comment (andrew032011) [inline backend/src/API/teamEventsAPI.ts:11]: Nit: remove `-`

**Review by andrew032011 — CHANGES_REQUESTED** [review]
I see you've picked up some changes here that are from outside of IDOL when you were formatting. Can you revert those?

**Comment by andrew032011** [comment]
I see a lot of `If the user is not a lead or admin, then an error is thrown.` How about using the `@throws` decorator on the docstring instead?

File: backend/src/API/teamEventsAPI.ts [hunk 3]
```diff
@@ -22,9 +28,20 @@ export const getAllTeamEvents = async (user: IdolMember): Promise<TeamEvent[]> =
...
+ * @returns a list of the current team events
+ */
 export const getAllTeamEventInfo = async (): Promise<TeamEventInfo[]> =>
   TeamEventsDao.getAllTeamEventInfo();
 
+/**
+ * Creates a team event from a TeamEventInfo object which contains the name,
+ * date, credits, hours, and whether it is a community event or not.
+ * @param teamEventInfo - the information about the team event
+ * @param user - the user creating the team event
```
Comment (andrew032011) [inline backend/src/API/teamEventsAPI.ts:42]: `@throws {PermissionsError}`?

**Review by andrew032011 — APPROVED** [review]

**Pull request merged**